app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
```

Comment:
the `raw_algorithm` method is required by the `Algorithmable` concern in the upstream gem. It is not in the controller class either in upstream or the 18F fork.

The `signature_opts` method used to return a hashref, but now it is just a callthrough to the `algorithm` method. I left the `signature_opts` method defined rather than replacing its usage because it seemed to me to help the readability of the code to know *what* the algorithm was for.